specs/FrameProcesssInfo.md

```diff
@@ -0,0 +1,540 @@
+interface ICoreWebView2Environment14 : ICoreWebView2Environment13 {
+  /// Gets a snapshot collection of `ProcessInfo`s corresponding to all currently
+  /// running processes associated with this `CoreWebView2Environment` except 
+  /// for crashpad process. This provide the same list of `ProcessInfo`s as 
+  /// what's provided in `GetProcessInfos`. Plus, this provide a list of associated
+  /// `FrameInfo`s which are actively running (showing UI elements) in the renderer
+  /// process. Check `AssociatedFrameInfos` for acquiring this detail information.
+  /// 
+  /// \snippet ProcessComponent.cpp GetProcessInfosWithDetails
+  HRESULT GetProcessInfosWithDetails([in] ICoreWebView2GetProcessInfosWithDetailsCompletedHandler* handler);
+}
+
+/// A continuation of the ICoreWebView2FrameInfo interface.
+[uuid(a7a7e150-e2ca-11ed-b5ea-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2FrameInfo2 : ICoreWebView2FrameInfo {
+  /// The parent `FrameInfo`. `ParentFrameInfo` will only be populated when obtained
+  /// via calling `CoreWebView2ProcessInfo.AssociatedFrameInfos`. 
+  /// `CoreWebView2FrameInfo` objects obtained via `CoreWebView2.ProcessFailed` will
+  /// always have a `null` `ParentFrameInfo`. This property is also `null` for the  
+  /// top most document in the WebView2 which has no parent frame.
+  /// Note that this `ParentFrameInfo` could be out of date as it's a snapshot.   
+  [propget] HRESULT ParentFrameInfo([out, retval] ICoreWebView2FrameInfo** frameInfo);
+  /// The unique identifier of the frame associated with the current `FrameInfo`.
+  /// It's the same kind of ID as with the `FrameId` in `CoreWebView2` and via 
+  /// `CoreWebView2Frame`. `FrameId` will only be populated when obtained
+  /// calling `CoreWebView2ProcessInfo.AssociatedFrameInfos`. 
+  /// `CoreWebView2FrameInfo` objects obtained via `CoreWebView2.ProcessFailed` will
+  /// always have an invalid frame Id 0.
+  /// Note that this `FrameId` could be out of date as it's a snapshot. 
+  [propget] HRESULT FrameId([out, retval] UINT32* id);
+  /// The frame kind of the frame.
+  [propget] HRESULT FrameKind([out, retval] COREWEBVIEW2_FRAME_KIND* kind);
+}
+
+/// A continuation of the ICoreWebView2Frame4 interface.
+[uuid(04baa798-a0e9-11ed-a8fc-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2Frame5: ICoreWebView2Frame4 {
+  /// The unique identifier of the current frame.
+  [propget] HRESULT FrameId([out, retval] UINT32* id);
+}
+
+/// A continuation of the ICoreWebView2_17 interface.
+[uuid(ad712504-a66d-11ed-afa1-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2_18 : ICoreWebView2_17 {
+  /// The unique identifier of the current frame.
+  /// Note that `FrameId` may not valid if `CoreWebView` has not done
+  /// any navigation. It's safe to get this value during or after the first
+  /// `DOMContentLoaded` event. Else, it could return an invalid frame Id 0.

```

Comment:
Is DOMContentLoaded really the first navigation event where its safe? DOMContentLoaded is when the document has been fully obtained from the network, parsed, and all synchronous javascript has finished running. Also, not sure that DOMContentLoaded is raised for things that aren't HTML documents like PDFs. I would have guessed NavigationStarting or ContentLoading would work better?

Reply:
I've tested `DOMContentLoaded` is also raised for other documents like PDF, JPG. 
`NavigationStarting` happens before than the place we pass the `FrameID`. 
While I take a closer look, `ContentLoading` can be the first safe navigation event for retrieving `FrameId`.   
I'm not see any exception that `CoreWebView2` bypass this event in the first navigation. 
From my reading about the [Same Documentation Navigation](https://chromium.googlesource.com/chromium/src/+/main/docs/navigation_concepts.md#same_document-and-cross_document-navigations) So I think your suggestion about `ContentLoading` is better choice.